specs/~dynamic-names.yml

```
@@ -0,0 +1,205 @@
+    desc: The dynamic partial should operate within the current context.
+    data: { text: 'Hello, world!', foo: { bar: { baz: 'partial' } } }
+    template: '"{{>*foo.bar.baz}}"'
+    partials: { partial: '*{{text}}*' }
+    expected: '"*Hello, world!*"'
+
+  - name: Dotted Names - Failed Lookup
+    desc: The dynamic partial should operate within the current context.
+    data:
+      text: 'Hello, world!'
+      foo: 'test'
+      test:
+        bar:
+          baz: 'partial'
+    template: '"{{>*foo.bar.baz}}"'
+    partials: { partial: '*{{text}}*' }
+    expected: '""'
+
+  - name: Dotted Names - Failed Lookup
+    desc: The dynamic partial should operate within the current context.
+    data:
+      foo:
+        text: 'Hello, world!'
+        bar:
+          baz: 'partial'
+    template: '"{{>*foo.bar.baz}}"'
+    partials: { partial: '*{{text}}*' }
+    expected: '"**"'
+
+  - name: Dotted names - Context Stacking
+    desc: Dotted names should not push a new frame on the context stack.
+    data:
+      section1: { value: 'section1' }
+      section2: { dynamic: 'partial', value: 'section2' }
+    template: "{{#section1}}{{>*section2.dynamic}}{{/section1}}"
+    partials:
+      partial: '"{{value}}"'
+    expected: '"section1"'
+
+  - name: Dotted names - Context Stacking Under Repetition
+    desc: Dotted names should not push a new frame on the context stack.
+    data:
+      value: 'test'
+      section1: [ 1, 2 ]
+      section2: { dynamic: 'partial', value: 'section2' }
+    template: "{{#section1}}{{>*section2.dynamic}}{{/section1}}"
+    partials:
+      partial: "{{value}}"
+    expected: "testtest"
+
+  - name: Dotted names - Context Stacking Failed Lookup
+    desc: Dotted names should resolve against the proper context stack.
+    data:
+      section1: [ 1, 2 ]
+      section2: { dynamic: 'partial', value: 'section2' }
+    template: "{{#section1}}{{>*section2.dynamic}}{{/section1}}"
+    partials:
+      partial: '"{{value}}"'
+    expected: '""""'
+
+  - name: Recursion
+    desc: Dynamic partials should properly recurse.
+    data:
+      template: 'node'
+      content: 'X'
+      nodes: [ { content: 'Y', nodes: [] } ]
+    template: '{{>*template}}'
+    partials: { node: '{{content}}<{{#nodes}}{{>*template}}{{/nodes}}>' }
+    expected: 'X<Y<>>'
+
+  # Whitespace Sensitivity
+
+  - name: Surrounding Whitespace
+    desc: |
+      A dynamic partials should not alter surrounding whitespace; any
+      whitespace preceding the tag should treated as indentation while any
+      whitepsace succeding the tag should be left untouched.
+    data: { partial: 'foobar' }
+    template: '| {{>*partial}} |'
+    partials: { foobar: "\t|\t" }
+    expected: "| \t|\t |"
+
+  - name: Inline Indentation
+    desc: |
+      Whitespace should be left untouched: whitespaces preceding the tag
+      should be treated as indentation.
+    data: { dynamic: 'partial', data: '|' }
+    template: "  {{data}}  {{>* dynamic}}\n"
```

I still feel this space doesn't belong here, because the test is about indentation rather than internal padding. The space can optionally be transplanted to the Padding Whitespace test.

```suggestion
    template: "  {{data}}  {{>*dynamic}}\n"
```